tests/test_modeling_flax_common.py

```diff
@@ -160,12 +160,58 @@ def recursive_check(tuple_object, dict_object):
             dict_inputs = self._prepare_for_class(inputs_dict, model_class)
             check_equivalence(model, tuple_inputs, dict_inputs, {"output_hidden_states": True})
 
+    def check_outputs(self, fxo, pto, model_class, names):
+        """
+        Args:
+            model_class: The class of the model that is currently testing. For example, ..., etc.
+            Currently unused, but it could make debugging easier and faster.
+
+            names: A string, or a list of strings. These specify what fxo/pto represent in the model outputs.
+                Currently unused, but in the future, we could use this information to make the error message clearer
+                by giving the name(s) of the output tensor(s) with large difference(s) between PT and Flax.
+        """
+        if type(fxo) == tuple:
+            self.assertEqual(type(pto), tuple)
+            self.assertEqual(len(fxo), len(pto))
+            if type(names) in [tuple, list]:
+                for fo, po, name in zip(fxo, pto, names):
+                    self.check_outputs(fo, po, model_class, names=name)
+            elif type(names) == str:
+                for idx, (fo, po) in enumerate(zip(fxo, pto)):
+                    self.check_outputs(fo, po, model_class, names=f"{names}_{idx}")
+        elif isinstance(fxo, jnp.ndarray):
+            self.assertTrue(isinstance(pto, torch.Tensor))
+
+            fxo = fxo.numpy()
+            pto = pto.numpy()
+
+            fx_nans = np.copy(np.isnan(fxo))
+            pt_nans = np.copy(np.isnan(pto))
+
+            pto[fx_nans] = 0
+            fxo[fx_nans] = 0
+            pto[pt_nans] = 0
+            fxo[pt_nans] = 0
+
+            max_diff = np.amax(np.abs(fxo - pto))
+            self.assertLessEqual(max_diff, 1e-5)
```

Comment:
I'm not sure if `1e-5` will work for all models especially on TPU/GPU since JAX does some approximations on TPU so the output can diverge. cf https://github.com/huggingface/transformers/issues/15754
What do you think @patrickvonplaten

Reply:
I will check on GPU VM - currently I am doing this for PT/TF.

Reply:
Yes I think a precision of `1e-3` would be better